internal/tool/blob.go

```diff
@@ -44,23 +49,48 @@ type OutputFetchBlob struct {
 
 // FetchBlob fetches blob referenced by a digest in a manifest.
 func FetchBlob(ctx context.Context, _ *mcp.CallToolRequest, input InputFetchBlob) (*mcp.CallToolResult, OutputFetchBlob, error) {
-	if input.Registry == "" || input.Repository == "" || input.Digest == "" {
-		return nil, OutputFetchBlob{}, fmt.Errorf("registry, repository, and digest are required")
+	// validate input
+	if input.Registry == "" {
+		return nil, OutputFetchBlob{}, fmt.Errorf("registry name is required")
 	}
+	if input.Repository == "" {
+		return nil, OutputFetchBlob{}, fmt.Errorf("repository name is required")
+	}
+	if input.Digest == "" {
+		return nil, OutputFetchBlob{}, fmt.Errorf("blob digest is required")
+	}
+	ref := registry.Reference{
+		Registry:   input.Registry,
+		Repository: input.Repository,
+		Reference:  input.Digest,
+	}
+	if err := ref.Validate(); err != nil {
+		return nil, OutputFetchBlob{}, err
+	}
+	repo := remote.NewRepository(ref)
 
-	// construct the reference string
-	reference := fmt.Sprintf("%s/%s@%s", input.Registry, input.Repository, input.Digest)
-
-	// fetch blob using oras CLI
-	cmd := exec.CommandContext(ctx, "oras", "blob", "fetch", "-o-", reference)
-	result, err := cmd.Output()
+	// fetch the blob
+	desc, rc, err := repo.Blobs().FetchReference(ctx, ref.Reference)
+	if err != nil {
+		return nil, OutputFetchBlob{}, err
+	}
+	defer rc.Close()
+	if desc.Size > maxBlobSize {
+		return nil, OutputFetchBlob{}, fmt.Errorf("blob too large: %d", desc.Size)
+	}
+	blobBytes, err := content.ReadAll(rc, desc)
```

Comment:
Should we wrap `rc` with a limited reader here?

Reply:
In fact, the `content.ReadAll` does the limit read job there.

Reply:
Sounds good then